codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyFrameCodec.java

```diff
@@ -125,13 +125,16 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
 
     @Override
     public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
+        resetReadState();
```

Comment:
What about adding a method to `ByteToMessageDecoder` like this:

```java
@Override
public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
  channelReadComplete(ctx, !decodeWasNull);
}

protected final void channelReadComplete(ChannelHandlerContext ctx, boolean readData) throws Exception {
        numReads = 0;
        discardSomeReadBytes();
        decodeWasNull = false;
        if (readData) {
          ctx.fireChannelReadComplete(); 
        } else if (ctx.channel().config().isAutoRead()) {
          ctx.read();
        }
}
```

Then in SpdyFrameCodec.java you can do this:

```java
public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
  // Not sure if it is necessary to reset before the call ... but just in case.
  boolean wasRead = read;
  read = false;
  channelReadComplete(ctx, wasRead);
}
```